### PR TITLE
Feat/add info improvements

### PR DESCRIPTION
**Pull request by sanjayankur31**

### CODE REVIEW

**Comment by sanjayankur31** [comment]
Example test script would be something like this:

```

import neuroml

net = neuroml.nml.nml.Network(id="net")
p = neuroml.nml.nml.Projection(id="p")
net.add(p)
print("one true")
net.info(True)

print("both true")
net.info(True, True)

izh = neuroml.nml.nml.Izhikevich2007Cell(id="i")
izh.info(show_all_contents=True)

```

**Comment by sanjayankur31** [comment]
@irahulsonkar take a look at this, and please do review it and note what else would be useful to add here.

**Comment by irahulsonkar** [comment]
> @irahulsonkar take a look at this, and please do review it and note what else would be useful to add here.

Thanks! I will check this.

**Comment by sanjayankur31** [comment]
You can see the updated method documentation here on the RTD pull request build too:

https://libneuroml--127.org.readthedocs.build/en/127/userdocs/coreclasses.html#neuroml.nml.nml.BaseWithoutId.info

(I think it should stay while the PR is open)

**Comment by sanjayankur31** [comment]
@pgleeson first fall outs from the schema changes:

- `sg3.members.append(Member(seg.id))` ->  `sg3.members.append(Member(segments=seg.id))` because now `Member` extends `BaseWithoutId` and so it's first argument in the constructor is `NeuroLexId`, so `segments=` is required here.
- Same for `Includes` and a number of other component types that now extend `BaseWithoutId` etc. 

So, since the API has changed here, this is a breaking change. I'll go over the tests and fix them here.

**Comment by sanjayankur31** [comment]
I've gone through the code and ensure that all ComponentType constructors (neuroml.* class constructors) use keyword arguments in the code instead of positional arguments to ensure that the correct members are set when constructors are used.

The primary issue is that the `neurolexid` argument is always first in the constructor definition, because it's inherited from the base classes. So if a user uses positional arguments, e.g.: `Location(10, 10, 10)`, what's being set is `neurolexid=10, x=10, y=10, z=None`. So, keyword arguments must be used to correctly instantiate the object: `Location(x=10, y=10, z=10)`.

I wonder if there's a way of pushing neurolexid to the end of the constructor argument list in generateds..

Should be ready for review now @pgleeson . I've marked the nml.py regenerations as breaking changes too, since the API has changed a little (to include neurolexid as a constructor argument for the updated classes).

**Comment by pgleeson** [comment]
Regarding `neurolexid`, there's probably not any reason to have this as an allowed attribute on anything unless we explicitly know it would be useful, like a cell/population/segment group which may correspond to a known physiological entity. Maybe the best solution is having a simpler base class without  `neurolexid` that things can extend from by default, and another including it for more "neurosciency" entities

**Comment by sanjayankur31** [comment]
I could remove NeurolexId from BaseWithoutId, and we could create a new one, say BaseWithNeurolexId, but I'm not sure where in the hierarchy this new one will fit. 

Can a complextype extend from multiple ancestors in XSD (I don't think so, but the docs are so huge that I haven't gone through them all)? If yes, then anything that needs NeurolexId can also extend the new class. If not, we may not be able to have a linear inheritance without something or the other having NeurolexId but not needing it?

I guess the most specific solution is to only add NeurolexId as an attribute to individual components that need it?

**Comment by sanjayankur31** [comment]
I've added it to specific component types for now. See: https://github.com/NeuroML/NeuroML2/pull/194

File: neuroml/nml/helper_methods.py [hunk 1]
```diff
@@ -1065,7 +1093,10 @@ def get_by_id(self,id):
             return None
         all_ids = []
         for ms in self.member_data_items_:
-            mlist = self.__getattribute__(ms.name)
+            mlist = getattr(self, ms.get_name())
+            # TODO: debug why this is required
+            if mlist is None:
```
Comment (sanjayankur31) [inline neuroml/nml/helper_methods.py:1098]: @pgleeson not sure why this is required, but without this, it doesn't continue iterating over the next elements of `self.member_data_items_`, just stops at the first one :worried:

**Pull request merged**